modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -176,17 +178,29 @@ object ConsensusManager {
           .ifM(internalFacilitateWith(EventTrigger.some), Applicative[F].unit)
     }
 
-    Spawn[F]
-      .start(clusterStorage.peerChanges.evalMap {
-        case (peerId, Some(peer)) if peer.state === NodeState.Observing =>
-          val res = for {
-            exchangeRequest <- consensusStorage.getOwnRegistration.map(RegistrationExchangeRequest(_))
-            exchangeResponse <- consensusClient.exchangeRegistration(exchangeRequest).run(peer)
-            _ <- exchangeResponse.maybeKey.traverse(consensusStorage.registerPeer(peerId))
-          } yield ()
-          res.handleErrorWith(err => logger.error(err)(s"Error exchanging consensus registration with peer ${peerId.show}"))
-        case _ => Applicative[F].unit
-      }.compile.drain)
-      .as(manager)
+    Spawn[F].start(
+      nodeStorage.nodeStates
+        .filter(NodeState.leaving.contains)
+        .evalTap { _ =>
+          consensusStorage.getLastKeyAndArtifact.map {
+            _.map { case (key, _) => key }
+              .map(Deregistration(_))
```

Comment:
I think it should be a `Order[Key].max(lastKey.next, ownRegistration)`. In most cases the `lastKey.next` would be sufficient, however when we're still observing, the `lastKey.next` is 1 ordinal less than `ownRegistration`.

Reply:
oh, I meant it to be `lastKey.next`. anyway, I'll make max of these two